app/services/auth/okta_service.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Auth
+  class OktaService < BaseService
+    def authorize(email:)
+      email_domain = email.split('@').last
+      okta_integration = ::Integrations::OktaIntegration
+        .where('settings->>\'domain\' IS NOT NULL')
+        .where('settings->>\'domain\' = ?', email_domain)
+        .first
+
+      return result.single_validation_failure!(error_code: 'domain_not_configured') if okta_integration.blank?
+
+      initialize_state
+
+      params = {
+        client_id: okta_integration.client_id,
+        response_type: 'code',
+        response_mode: 'query',
+        scope: 'openid profile email',
+        redirect_uri: "#{ENV['LAGO_FRONT_URL']}/auth/okta/callback",
+        state: CurrentContext.okta_state,
```

Comment:
Is there a way to pass this context in parameters / response, rather than relying on the context?

Reply:
I need to keep the state for the next call from front end to check if it's the same coming back from Okta, we can discuss it tomorrow!